app/jobs/import_versions_job.rb

```diff
@@ -24,8 +24,10 @@ def perform(import)
   end
 
   def import_raw_data(raw_data)
+    total_rows = raw_data.split("\n").length
     each_json_line(raw_data) do |record, row|
       begin
+        puts "Importing row #{row+1}/#{total_rows}..." if ((row+1) % 25 == 0)
```

Comment:
`puts` is *not ok* in this context. Please use `Rails.logger.[whatever log level is appropriate]`. This needs to be able to function properly for normal imports when the application is running.

FYI, the reason it’s written without a total was so that it would be easy to transition to a streaming, lower-memory model (which necessarily has the caveat that you don’t know how long the stream is until you are done). Most clients in practice are nice to the app and avoid imports so big that we’ve needed to solve that, though, so I guess this is ok.

Reply:
Also, the whole import can be a single JSON file instead of a newline-delimited JSON stream, so `raw_data.split('\n').length` is not an accurate count of records.

Reply:
Sorry, I was trying to admit with the "poor [man's] logging" commit msg that this was ugly and wrong, but just trying to show an aspiring user experience that I thought might be more friendly to show progress on 15+ minutes of silent output :)

I messed around with `Rails.logger.info` for 15 minutes, but it seemed to be shuttling output into `log/development.log` or some such place that wasn't obvious, so I thought someone who understood rails better could give the quick advice on how to do this right. Halp? (Any advice on custom loggers -- if that's the right approach -- would be sincerely appreciated!)